syntax = "proto3";
import "google/protobuf/empty.proto";

package com.piaspspd.grpc;

service ProductService {
  // Busca parcial
  rpc GetProducts (nameQuery) returns (stream Product);
  rpc GetProduct (nameQuery) returns (Product);

  rpc addProduct (Product) returns (google.protobuf.Empty);
  rpc addProducts (stream Product) returns (google.protobuf.Empty);

  rpc removeProduct(nameQuery) returns (google.protobuf.Empty);
  rpc updateProduct (Product) returns (google.protobuf.Empty);
}

message nameQuery {
  string name = 1;
}

message Product {
  string name = 1;
  float price = 2;
  int32 amount = 3; // Quantidade em estoque
}
